scripts/apps/authoring/authoring/directives/SendItem.tsx

```diff
@@ -824,8 +824,16 @@ export function SendItem($q,
                     return;
                 }
                 // set the desks for desk filter
-                if (scope.currentUserAction === ctrl.userActions.publish) {
+                if (scope.currentUserAction === ctrl.userActions.publish || !privileges.privileges.move) {
```

Comment:
Can you explain this code? It looks strange that a particular piece of code gets executed if action is `publish` or the user doesn't have a `move` privilege. What does publishing have to do with moving? Shouldn't these be independent actions?

Reply:
Publish action is a compound action consisting of publish and optional send actions. When a story gets published users could choose to send the published story to the desks she is a member of. But they could send an unpublished story to any desk. Now we'd like to limit the desks to member desks for sending the (unpublished) stories if user doesn't have the move privilege.

Reply:
The requirements make sense, but the implementation is confusing. Even if publishing is a compound action from a user perspective, can you separate it in the code?

Reply:
This function sets desks and stages for user to send the story to. There are two things that affect the list of desks and stages, one if the action is publish the other is if the user has the move privilege. I'm not sure if I need to separate it, but I can add some comments here. What do you think?

Reply:
Yeah, a comment might do for now. Your first reply explains it pretty well so I'd just use that.